homeassistant/components/airzone_cloud/entity.py

```diff
@@ -0,0 +1,54 @@
+"""Entity classes for the Airzone Cloud integration."""
+from __future__ import annotations
+
+from typing import Any
+
+from aioairzone_cloud.const import AZD_NAME, AZD_SYSTEM_ID, AZD_ZONES
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.helpers.entity import DeviceInfo
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from .const import DOMAIN, MANUFACTURER
+from .coordinator import AirzoneUpdateCoordinator
+
+
+class AirzoneEntity(CoordinatorEntity[AirzoneUpdateCoordinator]):
+    """Define an Airzone Cloud entity."""
+
+    def get_airzone_value(self, key: str) -> Any:
+        """Return Airzone Cloud entity value by key."""
+        raise NotImplementedError()
+
+
+class AirzoneZoneEntity(AirzoneEntity):
+    """Define an Airzone Cloud Zone entity."""
+
+    def __init__(
+        self,
+        coordinator: AirzoneUpdateCoordinator,
+        entry: ConfigEntry,
+        zone_id: str,
+        zone_data: dict[str, Any],
+    ) -> None:
+        """Initialize."""
+        super().__init__(coordinator)
+
+        self.system_id = zone_data[AZD_SYSTEM_ID]
+        self.zone_id = zone_id
+
+        self._attr_device_info = DeviceInfo(
+            identifiers={(DOMAIN, f"{entry.entry_id}_{zone_id}")},
+            manufacturer=MANUFACTURER,
+            name=zone_data[AZD_NAME],
+            via_device=(DOMAIN, f"{entry.entry_id}_{self.system_id}"),
+        )
+        self._attr_unique_id = entry.unique_id or entry.entry_id
```

Comment:
Why is the unique_id set here, and then overwritten in the derived class `AirzoneZoneSensor`?

Reply:
You're right, I copied this from the Airzone Local integration and I think that it can be removed there too:
https://github.com/home-assistant/core/blob/657d285e8fe346908a1eb618dd4d987bb3190236/homeassistant/components/airzone/entity.py#LL131C14-L131C29